### PR TITLE
Bug preprocess translate

### PR DESCRIPTION
**Pull request by jmcrego**

### CODE REVIEW

File: translate.lua [hunk 1]
```diff
@@ -111,11 +111,32 @@ local function main()
...
       end
       tokenizers[idx][k] = v
-      if k == "bpe_model" and v ~= '' then
-      bpes[idx] = BPE.new(v)
-    end
     end
   end
+
+  if opt.tok_src_bpe_model ~= '' then
+     myopt = {}
```
Comment (guillaumekln) [inline translate.lua]: Should be:

```lua
local myopt = {}
```

File: translate.lua [hunk 2]
```diff
@@ -111,11 +111,32 @@ local function main()
...
+     myopt.bpe_EOT_marker = opt.tok_src_bpe_EOT_marker
+     myopt.bpe_BOT_marker = opt.tok_src_bpe_BOT_marker
+     myopt.joiner_new = opt.tok_src_joiner_new
+     myopt.joiner_annotate = opt.tok_src_joiner_annotate
+     myopt.bpe_mode = opt.tok_src_bpe_mode
+     myopt.bpe_case_insensitive = opt.tok_src_bpe_case_insensitive
+     bpes[1] = BPE.new(myopt)
+  end
+  if opt.tok_tgt_bpe_model ~= '' then
+     myopt = {}
```
Comment (guillaumekln) [inline translate.lua]: Same here.

**Comment by shrutijpalaskar** [comment]
Hi, the new translate.lua file now fails. Following is my command and the error log:
```
th translate.lua -src ../how-to/data/final_cv.txt -tgt ../how-to/data/final_char_labels.cv_arranged_nouttid -idx_files true -batch_size 1 -model $MODEL -output $OUTNAME -gpuid 1 

[10/16/17 11:22:23 INFO] Using GPU(s): 1
[10/16/17 11:22:23 WARNING] The caching CUDA memory allocator is enabled. This allocator improves performance at the cost of a higher GPU memory usage. To optimize for memory, consider disabling it by setting the environment variable: THC_CACHING_ALLOCATOR=0
[10/16/17 11:22:23 INFO] Loading '../how-to/models/cont_90hr_chars_inconcat_ad1_pdbrnn_E3_D2_dense_nonlinear_RNN512_EWV120_DWV30_sgd0.0861_naive0.3_bs32_curr_decat1_thres0.1_epoch8_1.39.t7'...
[10/16/17 11:22:27 INFO] Model seq2seq trained on feattext
[10/16/17 11:22:27 INFO] Using on-the-fly 'space' tokenization for input 1
[10/16/17 11:22:27 INFO] Using on-the-fly 'space' tokenization for input 2
/home/palaskar/torch-lua5.2/install/bin/lua: ...alaskar/torch-lua5.2/install/share/lua/5.2/nn/Linear.lua:66: size mismatch at /home/palaskar/torch-lua5.2/extra/cutorch/lib/THC/generic/THCTensorMathBlas.cu:243
stack traceback:
	[C]: in function 'addmm'
	...alaskar/torch-lua5.2/install/share/lua/5.2/nn/Linear.lua:66: in function <...alaskar/torch-lua5.2/install/share/lua/5.2/nn/Linear.lua:53>
	(...tail calls...)
	...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:345: in function 'neteval'
	...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:380: in function <...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:300>
	(...tail calls...)
	...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:345: in function 'neteval'
	...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:380: in function <...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:300>
	(...tail calls...)
	./onmt/modules/Network.lua:16: in function <./onmt/modules/Network.lua:15>
	(...tail calls...)
	...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:345: in function 'neteval'
	...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:380: in function <...r/torch-lua5.2/install/share/lua/5.2/nngraph/gmodule.lua:300>
	(...tail calls...)
	./onmt/modules/Encoder.lua:240: in function 'forward'
	./onmt/modules/BiEncoder.lua:153: in function 'forward'
	./onmt/modules/PDBiEncoder.lua:137: in function 'forward'
	./onmt/translate/Translator.lua:316: in function 'translateBatch'
	./onmt/translate/Translator.lua:512: in function 'translate'
	translate.lua:211: in function 'main'
	translate.lua:327: in main chunk
	[C]: in function 'dofile'
	...rch-lua5.2/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: in ?

**Comment by guillaumekln** [comment]
Hello,

This does not seem related to this particular PR. Could you open an issue instead?

Thanks.

**Comment by shrutijpalaskar** [comment]
Sorry for the mistake! I have opened a new issue. Thanks!

**Pull request merged**